pkg/adaptor/cloud/azure/provider.go

```diff
@@ -149,15 +149,40 @@ func (p *azureProvider) createNetworkInterface(ctx context.Context, nicName stri
 
 func (p *azureProvider) CreateInstance(ctx context.Context, podName, sandboxID string, cloudConfig cloudinit.CloudConfigGenerator, spec cloud.InstanceTypeSpec) (*cloud.Instance, error) {
 
+	var b64EncData string
+
 	instanceName := util.GenerateInstanceName(podName, sandboxID, maxInstanceNameLen)
 
-	userData, err := cloudConfig.Generate()
+	cloudConfigData, err := cloudConfig.Generate()
 	if err != nil {
 		return nil, err
 	}
 
-	//Convert userData to base64
-	userDataEnc := base64.StdEncoding.EncodeToString([]byte(userData))
+	// Copy the data in {} after content: from customData
+	/*
+		#cloud-config
+		write_files:
+		  - path: /peerpod/daemon.json
+		    content: |
+		      {
+		       ...
+			  }
+		  - path: other files
+	*/
+
+	if !p.serviceConfig.DisableCloudConfig {
+		//Convert cloudConfigData to base64
+		b64EncData = base64.StdEncoding.EncodeToString([]byte(cloudConfigData))
+	} else {
+		userData := strings.Split(cloudConfigData, "content: |")[1]
```

Comment:
Maybe i missed that, is there size limit for userdata passed using imds? is it validated? IIRC data size is checked, might be different/irrelevant now

Reply:
There is indeed a limit to userData - 64KB. I think it's better to check the size and throw error if the size is more than 64KB.
I'll add the check. Thanks @snir911

Reply:
I have added the explicit check for Azure. As per the Azure doc, both user data (via metadata service) or custom data (via config drive) cannot exceed 64KB. This check was not present in the original code as well. So I added it as a separate commit.